src/main/java/com/internet/shop/dao/mysqlimpl/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,147 @@
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create user " + user, e);
+        }
+        return user;
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users WHERE user_id=?;";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getUserFromResultSet(resultSet, connection));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get user by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        String query = "SELECT * FROM users WHERE deleted = FALSE";
+        List<User> users = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                users.add(getUserFromResultSet(resultSet, connection));
+            }
+            return users;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get all users ", e);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE users SET user_name=?, login=?"
+                    + "WHERE password=? & deleted=FALSE";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update user " + user, e);
+        }
+        return user;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE users SET deleted=TRUE WHERE user_id=?;";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            String queryDel = "DELETE FROM shopping_carts_products WHERE cart_id=?";
```

Comment:
dao layer doesn't contain business logic, so let's remove deleting for the cart from here
besides not every user should have shoppingcart